### PR TITLE
test: parse providers from env vars

### PR DESCRIPTION
**Pull request by blaklafter**
## 📑 Description

Added some tests around parsing providers from env vars.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

There might be some behavior when parsing a malformed env var that is unintentional. See comments...

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -87,3 +88,93 @@ def test_parse_with_alert_source_with_no_providers_file():
...
+
+        #ASSERT
+        expected = {provider_name.replace("_", "-").lower(): provider_dict}
+        
+        # This might be a bug?
+        # It will create a provider context with an empty string as a provider name: {'': {'hi': 1}}
+        assert parser.context_manager.providers_context == expected
+
+        # I would expect it to not create the provider
+        #assert parser.context_manager.providers_context == {}
```
Comment (blaklafter) [inline tests/test_parser.py:176]: This might not be intentional. I wanted to draw attention to it as a corner case.

**Review by talboren — APPROVED** [review]
LGTM, thank you for that!

File: tests/test_parser.py (hunk 1 above)
Comment (talboren) [reply tests/test_parser.py]: i.e. a "nameless" provider. That is an edge case but probably not to interesting to tackle right now.
